doc/examples/particles/lib/main.dart

```diff
@@ -0,0 +1,401 @@
+
+import 'package:flame/components/particle_component.dart';
+import 'package:flame/components/particles/circle_particle.dart';
+import 'package:flame/components/particles/moving_particle.dart';
+import 'package:flame/components/particles/translated_particle.dart';
+import 'package:flame/components/particles/computed_particle.dart';
+import 'package:flame/components/particles/image_particle.dart';
+import 'package:flame/components/particles/rotating_particle.dart';
+import 'package:flame/components/particles/accelerated_particle.dart';
+import 'package:flame/components/particles/paint_particle.dart';
+import 'package:flame/flame.dart';
+import 'package:flame/game.dart';
+import 'package:flame/position.dart';
+import 'package:flame/text_config.dart';
+import 'package:flutter/material.dart';
+
+void main() async {
+  Size gameSize;
+  WidgetsFlutterBinding.ensureInitialized();
+
+  await Future.wait([
+    Flame.util.initialDimensions().then((size) => gameSize = size),
+    Flame.images.loadAll(const ['zap.png']),
+  ]);
+
+  final game = MyGame(gameSize);
+  runApp(game.widget);
+}
+
+class MyGame extends BaseGame {
+  /// Defines dimensions of the sample
+  /// grid to be displayed on the screen,
+  /// 5x5 in this particular case
+  static const gridSize = 5;
+  static const steps = 5;
+
+  /// Miscellaneous values used
+  /// by examples below
+  final Random rnd = Random();
+  final StepTween steppedTween = StepTween(begin: 0, end: 5);
+  final TextConfig fpsTextConfig = const TextConfig(
+    color: const Color(0xFFFFFFFF),
+  );
+
+  Offset cellSize;
+  Offset halfCellSize;
+
+  MyGame(Size screenSize) {
+    size = screenSize;
+    cellSize = Offset(size.width / gridSize, size.height / gridSize);
+    halfCellSize = cellSize * .5;
+
+    // Spawn new particles every second
+    Timer.periodic(const Duration(seconds: 1), (_) => spawnParticles());
+  }
+
+  /// Showcases various different uses of [Particle]
+  /// and its derivatives
+  void spawnParticles() {
+    // Contains sample particles, in order by complexity
+    // and amount of used features. Jump to source for more explanation on each
+    final particles = <Particle>[
+      circle(),
+      smallWhiteCircle(),
+      movingParticle(),
+      randomMovingParticle(),
+      alignedMovingParticles(),
+      easedMovingParticle(),
+      intervalMovingParticle(),
+      computedParticle(),
+      steppedComputedParticle(),
+      reuseParticles(),
+      imageParticle(),
+      reuseImageParticle(),
+      rotatingImage(),
+      acceleratedParticles(),
+      paintParticle(),
+    ];
+
+    // Place all the [Particle] instances
+    // defined above in a grid on the screen
+    // as per defined grid parameters
+    do {
+      final particle = particles.removeLast();
+      final col = particles.length % gridSize;
+      final row = particles.length ~/ gridSize;
+      final cellCenter =
+          cellSize.scale(col.toDouble(), row.toDouble()) + (cellSize * .5);
+
+      add(
```

Comment:
you should use `addLater` here because this might be running in parallel with the update loop

Reply:
Actually, if my understanding of microtasks in Dart is correct, it's already decoupled from the frame which runs the game loop update by being within a `Timer` which is separate from `Flame` game loop. But it wouldn't hurt to add either 😂.

Reply:
is it not possible they could run in separate threads and race each other? I never used the Timer like that so I don't really know, but for click events, I know it can happen

Reply:
Yup, the tasks from `Event` queue are processed one by one in a single thread so they would not interfere with each other. That is very similar on how threading works in JS too. So we have only one thread and there's only one line of code is executed at any given moment of time. So it's not possible that something would interject right in the middle of the game frame and would mess the components.